### ISSUE TITLE
As a developer reviewing a pull request, I want to have the build status show pending until the build has completed so I don't accidentally merge a branch which has not passed build

### ISSUE THREAD

**Issue by mrbelvedere** [SalesforceFoundation/Cumulus#1065]
The GitHub Commit Status API allows for a "pending" status.  The implementation would be to add an additional signal handler on the creation of a pull request in mrbelvedere that would set the build status to pending on any new pull request in the repository and on any new commit on a branch with an open pull request.  This way, the pull request would clearly show that a build is pending rather than showing no build status at all.
